pkg/skaffold/cluster/minikube.go

```diff
@@ -0,0 +1,215 @@
+		return true
+	}
+
+	if ok, err := matchProfileAndServerURL(kubeContext); err != nil {
+		logrus.Tracef("failed to match minikube profile: %v", err)
+	} else if ok {
+		logrus.Debugf("Minikube cluster detected: context %q has matching profile name or server url", kubeContext)
+		return true
+	}
+	logrus.Tracef("Minikube cluster not detected for context %q", kubeContext)
+	return false
+}
+
+func (clientImpl) MinikubeExec(arg ...string) (*exec.Cmd, error) {
+	return minikubeExec(arg...)
+}
+
+func minikubeExec(arg ...string) (*exec.Cmd, error) {
+	b, err := minikubeBinaryFunc()
+	if err != nil {
+		return nil, fmt.Errorf("getting minikube executable: %w", err)
+	}
+	return exec.Command(b, arg...), nil
+}
+
+func minikubeBinary() (string, error) {
+	execName := "minikube"
+	if found, _ := util.DetectWSL(); found {
+		execName = "minikube.exe"
+	}
+	filename, err := exec.LookPath(execName)
+	if err != nil {
+		return "", errors.New("unable to find minikube executable. Please add it to PATH environment variable")
+	}
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return "", fmt.Errorf("unable to find minikube executable. File not found %s", filename)
+	}
+	return filename, nil
+}
+
+// matchClusterCertPath checks if the cluster certificate for this context is from inside the minikube directory
+func matchClusterCertPath(kubeContext string) (bool, error) {
+	c, err := getClusterInfo(kubeContext)
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes config: %w", err)
+	}
+	if c.CertificateAuthority == "" {
+		return false, nil
+	}
+	return util.IsSubPath(minikubePath(), c.CertificateAuthority), nil
+}
+
+// matchProfileAndServerURL checks if kubecontext matches any valid minikube profile
+// and for selected drivers if the k8s server url is same as any of the minikube nodes IPs
+func matchProfileAndServerURL(kubeContext string) (bool, error) {
+	config, err := getRestClientConfigFunc()
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes config: %w", err)
+	}
+	apiServerURL, _, err := rest.DefaultServerURL(config.Host, config.APIPath, schema.GroupVersion{}, false)
+
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes server url: %w", err)
+	}
+
+	logrus.Tracef("kubernetes server url: %s", apiServerURL)
+
+	ok, err := matchServerURLFor(kubeContext, apiServerURL)
+	if err != nil {
+		return false, fmt.Errorf("checking minikube node url: %w", err)
+	}
+	return ok, nil
+}
+
+func matchServerURLFor(kubeContext string, serverURL *url.URL) (bool, error) {
+	cmd, err := minikubeExec("profile", "list", "-o", "json")
+	if err != nil {
+		return false, fmt.Errorf("executing minikube command: %w", err)
+	}
+
+	out, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return false, fmt.Errorf("getting minikube profiles: %w", err)
+	}
+
+	var data data
+	if err = json.Unmarshal(out, &data); err != nil {
+		log.Fatal(fmt.Errorf("failed to unmarshal data: %w", err))
```

Comment:
could not unmarshal `minikube profile list` output.
Fatal will cause the code to stop. 
Did you mean use to trace instead?

Reply:
my bad, definitely intended to use trace

Reply:
Done